File: system-test/typescript/src/testkit/dsl/port/myShop/scenario-dsl.ts
import type { AssumeStage } from './assume/assume-stage.js';
import type { GivenStage } from './given/given-stage.js';
import type { WhenStage } from './when/when-stage.js';

export interface ScenarioDsl {
  assume(): AssumeStage;
  given(): GivenStage;
  when(): WhenStage;
  close(): Promise<void>;
}

File: system-test/java/src/main/java/com/mycompany/myshop/testkit/dsl/port/ScenarioDsl.java
package com.mycompany.myshop.testkit.dsl.port;

import com.mycompany.myshop.testkit.dsl.port.assume.AssumeStage;
import com.mycompany.myshop.testkit.dsl.port.given.GivenStage;
import com.mycompany.myshop.testkit.dsl.port.when.WhenStage;

public interface ScenarioDsl {
    AssumeStage assume();

    GivenStage given();

    WhenStage when();
}